src/w3d/lib/coord.h

```diff
@@ -22,104 +22,66 @@ using GameMath::Sqrt;
 class Coord2D
 {
 public:
-    Coord2D() : x(0.0f), y(0.0f) {}
```

Comment:
Revert this is much safer

Reply:
What is the reasoning here to remove the default constructor? Is Coord2D meant to be an aggregate type? I doubt this is for performance. Keep in mind omitting default initialization in structs always carries the risk of non initialized values and subtle bugs.

On the other hand Vector3 & co also have no default constructor, so this seems consistent.

I would like this and similar classes to have a constructor that takes a ZERO enum and then initializes to zero. This way you can easily create object like this:

`Coord3D coord(ZERO);`

Reply:
I disagree with bringing back constructors in this class and think initialization should be explicit.